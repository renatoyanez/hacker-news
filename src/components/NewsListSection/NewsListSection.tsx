import { ReactNode, ReactElement } from 'react';
import styles from './newsListSection.module.css';

const NewsListSection = ({
	children,
}: {
	children?: ReactNode | ReactNode[];
}): ReactElement => {
	return (
		<div className={styles.container}>
			<section className={styles.cards}>
				<article className={styles.card}>card</article>
				<article className={styles.card}>card</article>
				<article className={styles.card}>card</article>
				<article className={styles.card}>card</article>
				<article className={styles.card}>card</article>

				{/* {children} */}
			</section>
		</div>
	);
};

export default NewsListSection;
