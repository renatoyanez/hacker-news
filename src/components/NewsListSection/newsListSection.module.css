.container {
	border: 1px solid blue;
}

.cards {
	display: flex;
	margin: 0 auto;
	justify-content: space-between;
	flex-wrap: wrap;
	/* width: 75%; */
}

.card {
	border: 2px solid black;
	border-radius: 10px;
	overflow: hidden;
	height: 400px;
	min-width: 300px;
	max-width: 400px;
	margin: 0 3px 1em 3px;
	flex: 0 1 25%;
}
