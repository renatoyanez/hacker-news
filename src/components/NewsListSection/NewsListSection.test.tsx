import React from 'react';
import { create } from 'react-test-renderer';
import NewsListSection, {
	NewsListSectionProps,
} from './NewsListSection';

const incrementNewsArray = (num: number) => {
	const contentArray = [];
	for (let i = 0; i < num; i++) {
		contentArray.push({
			objectID: `${i}`,
			created_at: `199${i}-${i}-${i}`,
			author: `author-${i}`,
			story_title: `title-${i}`,
			story_url: `https://www.${i}.com/`,
		});
	}
	return contentArray;
};

const props: NewsListSectionProps = {
	news: incrementNewsArray(10),
};

describe('NewsListSection', () => {
	it('renders correctly', () => {
		const tree = create(<NewsListSection {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
