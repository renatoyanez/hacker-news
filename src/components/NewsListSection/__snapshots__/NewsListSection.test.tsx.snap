// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsListSection renders correctly 1`] = `
<div>
  <section
    className="cards"
  >
    <article
      className="card"
    >
      <div
        className="cardContent"
      >
        <div
          className="infoSection"
        >
          <a
            href="https://www.0.com/"
            style={
              Object {
                "textDecoration": "none",
              }
            }
            target="_blank"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "margin": "0.1rem 0.5rem 0 0",
                  }
                }
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1.333c3.676 0 6.667 2.991 6.667 6.667 0 3.676-2.991 6.667-6.667 6.667-3.676 0-6.667-2.991-6.667-6.667 0-3.676 2.991-6.667 6.667-6.667zM8 0C3.582 0 0 3.582 0 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zm.667 8V4H7.333v5.333H12V8H8.667z"
                    fill="#606060"
                  />
                </svg>
              </div>
              <p
                className="textStyle6"
              >
                33 years ago by author-0
              </p>
            </div>
            <p
              className="textStyle3"
            >
              title-0
            </p>
          </a>
        </div>
        <div
          className="likeSection"
          data-testid="like-icon"
          onClick={[Function]}
        >
          <svg
            height="22"
            viewBox="0 0 24 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8.229C12.234 7.109 13.547 2 17.382 2 19.602 2 22 3.551 22 7.003c0 3.907-3.627 8.47-10 12.629C5.627 15.473 2 10.91 2 7.003c0-3.484 2.369-5.005 4.577-5.005 3.923 0 5.145 5.126 5.423 6.231zM0 7.003C0 11.071 3.06 16.484 12 22c8.94-5.516 12-10.929 12-14.997 0-7.962-9.648-9.028-12-3.737C9.662-1.996 0-1.004 0 7.003z"
              fill="#DD0031"
            />
          </svg>
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <div
        className="cardContent"
      >
        <div
          className="infoSection"
        >
          <a
            href="https://www.1.com/"
            style={
              Object {
                "textDecoration": "none",
              }
            }
            target="_blank"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "margin": "0.1rem 0.5rem 0 0",
                  }
                }
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1.333c3.676 0 6.667 2.991 6.667 6.667 0 3.676-2.991 6.667-6.667 6.667-3.676 0-6.667-2.991-6.667-6.667 0-3.676 2.991-6.667 6.667-6.667zM8 0C3.582 0 0 3.582 0 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zm.667 8V4H7.333v5.333H12V8H8.667z"
                    fill="#606060"
                  />
                </svg>
              </div>
              <p
                className="textStyle6"
              >
                32 years ago by author-1
              </p>
            </div>
            <p
              className="textStyle3"
            >
              title-1
            </p>
          </a>
        </div>
        <div
          className="likeSection"
          data-testid="like-icon"
          onClick={[Function]}
        >
          <svg
            height="22"
            viewBox="0 0 24 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8.229C12.234 7.109 13.547 2 17.382 2 19.602 2 22 3.551 22 7.003c0 3.907-3.627 8.47-10 12.629C5.627 15.473 2 10.91 2 7.003c0-3.484 2.369-5.005 4.577-5.005 3.923 0 5.145 5.126 5.423 6.231zM0 7.003C0 11.071 3.06 16.484 12 22c8.94-5.516 12-10.929 12-14.997 0-7.962-9.648-9.028-12-3.737C9.662-1.996 0-1.004 0 7.003z"
              fill="#DD0031"
            />
          </svg>
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <div
        className="cardContent"
      >
        <div
          className="infoSection"
        >
          <a
            href="https://www.2.com/"
            style={
              Object {
                "textDecoration": "none",
              }
            }
            target="_blank"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "margin": "0.1rem 0.5rem 0 0",
                  }
                }
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1.333c3.676 0 6.667 2.991 6.667 6.667 0 3.676-2.991 6.667-6.667 6.667-3.676 0-6.667-2.991-6.667-6.667 0-3.676 2.991-6.667 6.667-6.667zM8 0C3.582 0 0 3.582 0 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zm.667 8V4H7.333v5.333H12V8H8.667z"
                    fill="#606060"
                  />
                </svg>
              </div>
              <p
                className="textStyle6"
              >
                31 years ago by author-2
              </p>
            </div>
            <p
              className="textStyle3"
            >
              title-2
            </p>
          </a>
        </div>
        <div
          className="likeSection"
          data-testid="like-icon"
          onClick={[Function]}
        >
          <svg
            height="22"
            viewBox="0 0 24 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8.229C12.234 7.109 13.547 2 17.382 2 19.602 2 22 3.551 22 7.003c0 3.907-3.627 8.47-10 12.629C5.627 15.473 2 10.91 2 7.003c0-3.484 2.369-5.005 4.577-5.005 3.923 0 5.145 5.126 5.423 6.231zM0 7.003C0 11.071 3.06 16.484 12 22c8.94-5.516 12-10.929 12-14.997 0-7.962-9.648-9.028-12-3.737C9.662-1.996 0-1.004 0 7.003z"
              fill="#DD0031"
            />
          </svg>
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <div
        className="cardContent"
      >
        <div
          className="infoSection"
        >
          <a
            href="https://www.3.com/"
            style={
              Object {
                "textDecoration": "none",
              }
            }
            target="_blank"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "margin": "0.1rem 0.5rem 0 0",
                  }
                }
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1.333c3.676 0 6.667 2.991 6.667 6.667 0 3.676-2.991 6.667-6.667 6.667-3.676 0-6.667-2.991-6.667-6.667 0-3.676 2.991-6.667 6.667-6.667zM8 0C3.582 0 0 3.582 0 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zm.667 8V4H7.333v5.333H12V8H8.667z"
                    fill="#606060"
                  />
                </svg>
              </div>
              <p
                className="textStyle6"
              >
                30 years ago by author-3
              </p>
            </div>
            <p
              className="textStyle3"
            >
              title-3
            </p>
          </a>
        </div>
        <div
          className="likeSection"
          data-testid="like-icon"
          onClick={[Function]}
        >
          <svg
            height="22"
            viewBox="0 0 24 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8.229C12.234 7.109 13.547 2 17.382 2 19.602 2 22 3.551 22 7.003c0 3.907-3.627 8.47-10 12.629C5.627 15.473 2 10.91 2 7.003c0-3.484 2.369-5.005 4.577-5.005 3.923 0 5.145 5.126 5.423 6.231zM0 7.003C0 11.071 3.06 16.484 12 22c8.94-5.516 12-10.929 12-14.997 0-7.962-9.648-9.028-12-3.737C9.662-1.996 0-1.004 0 7.003z"
              fill="#DD0031"
            />
          </svg>
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <div
        className="cardContent"
      >
        <div
          className="infoSection"
        >
          <a
            href="https://www.4.com/"
            style={
              Object {
                "textDecoration": "none",
              }
            }
            target="_blank"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "margin": "0.1rem 0.5rem 0 0",
                  }
                }
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1.333c3.676 0 6.667 2.991 6.667 6.667 0 3.676-2.991 6.667-6.667 6.667-3.676 0-6.667-2.991-6.667-6.667 0-3.676 2.991-6.667 6.667-6.667zM8 0C3.582 0 0 3.582 0 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zm.667 8V4H7.333v5.333H12V8H8.667z"
                    fill="#606060"
                  />
                </svg>
              </div>
              <p
                className="textStyle6"
              >
                29 years ago by author-4
              </p>
            </div>
            <p
              className="textStyle3"
            >
              title-4
            </p>
          </a>
        </div>
        <div
          className="likeSection"
          data-testid="like-icon"
          onClick={[Function]}
        >
          <svg
            height="22"
            viewBox="0 0 24 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8.229C12.234 7.109 13.547 2 17.382 2 19.602 2 22 3.551 22 7.003c0 3.907-3.627 8.47-10 12.629C5.627 15.473 2 10.91 2 7.003c0-3.484 2.369-5.005 4.577-5.005 3.923 0 5.145 5.126 5.423 6.231zM0 7.003C0 11.071 3.06 16.484 12 22c8.94-5.516 12-10.929 12-14.997 0-7.962-9.648-9.028-12-3.737C9.662-1.996 0-1.004 0 7.003z"
              fill="#DD0031"
            />
          </svg>
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <div
        className="cardContent"
      >
        <div
          className="infoSection"
        >
          <a
            href="https://www.5.com/"
            style={
              Object {
                "textDecoration": "none",
              }
            }
            target="_blank"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "margin": "0.1rem 0.5rem 0 0",
                  }
                }
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1.333c3.676 0 6.667 2.991 6.667 6.667 0 3.676-2.991 6.667-6.667 6.667-3.676 0-6.667-2.991-6.667-6.667 0-3.676 2.991-6.667 6.667-6.667zM8 0C3.582 0 0 3.582 0 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zm.667 8V4H7.333v5.333H12V8H8.667z"
                    fill="#606060"
                  />
                </svg>
              </div>
              <p
                className="textStyle6"
              >
                28 years ago by author-5
              </p>
            </div>
            <p
              className="textStyle3"
            >
              title-5
            </p>
          </a>
        </div>
        <div
          className="likeSection"
          data-testid="like-icon"
          onClick={[Function]}
        >
          <svg
            height="22"
            viewBox="0 0 24 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8.229C12.234 7.109 13.547 2 17.382 2 19.602 2 22 3.551 22 7.003c0 3.907-3.627 8.47-10 12.629C5.627 15.473 2 10.91 2 7.003c0-3.484 2.369-5.005 4.577-5.005 3.923 0 5.145 5.126 5.423 6.231zM0 7.003C0 11.071 3.06 16.484 12 22c8.94-5.516 12-10.929 12-14.997 0-7.962-9.648-9.028-12-3.737C9.662-1.996 0-1.004 0 7.003z"
              fill="#DD0031"
            />
          </svg>
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <div
        className="cardContent"
      >
        <div
          className="infoSection"
        >
          <a
            href="https://www.6.com/"
            style={
              Object {
                "textDecoration": "none",
              }
            }
            target="_blank"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "margin": "0.1rem 0.5rem 0 0",
                  }
                }
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1.333c3.676 0 6.667 2.991 6.667 6.667 0 3.676-2.991 6.667-6.667 6.667-3.676 0-6.667-2.991-6.667-6.667 0-3.676 2.991-6.667 6.667-6.667zM8 0C3.582 0 0 3.582 0 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zm.667 8V4H7.333v5.333H12V8H8.667z"
                    fill="#606060"
                  />
                </svg>
              </div>
              <p
                className="textStyle6"
              >
                27 years ago by author-6
              </p>
            </div>
            <p
              className="textStyle3"
            >
              title-6
            </p>
          </a>
        </div>
        <div
          className="likeSection"
          data-testid="like-icon"
          onClick={[Function]}
        >
          <svg
            height="22"
            viewBox="0 0 24 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8.229C12.234 7.109 13.547 2 17.382 2 19.602 2 22 3.551 22 7.003c0 3.907-3.627 8.47-10 12.629C5.627 15.473 2 10.91 2 7.003c0-3.484 2.369-5.005 4.577-5.005 3.923 0 5.145 5.126 5.423 6.231zM0 7.003C0 11.071 3.06 16.484 12 22c8.94-5.516 12-10.929 12-14.997 0-7.962-9.648-9.028-12-3.737C9.662-1.996 0-1.004 0 7.003z"
              fill="#DD0031"
            />
          </svg>
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <div
        className="cardContent"
      >
        <div
          className="infoSection"
        >
          <a
            href="https://www.7.com/"
            style={
              Object {
                "textDecoration": "none",
              }
            }
            target="_blank"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "margin": "0.1rem 0.5rem 0 0",
                  }
                }
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1.333c3.676 0 6.667 2.991 6.667 6.667 0 3.676-2.991 6.667-6.667 6.667-3.676 0-6.667-2.991-6.667-6.667 0-3.676 2.991-6.667 6.667-6.667zM8 0C3.582 0 0 3.582 0 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zm.667 8V4H7.333v5.333H12V8H8.667z"
                    fill="#606060"
                  />
                </svg>
              </div>
              <p
                className="textStyle6"
              >
                26 years ago by author-7
              </p>
            </div>
            <p
              className="textStyle3"
            >
              title-7
            </p>
          </a>
        </div>
        <div
          className="likeSection"
          data-testid="like-icon"
          onClick={[Function]}
        >
          <svg
            height="22"
            viewBox="0 0 24 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8.229C12.234 7.109 13.547 2 17.382 2 19.602 2 22 3.551 22 7.003c0 3.907-3.627 8.47-10 12.629C5.627 15.473 2 10.91 2 7.003c0-3.484 2.369-5.005 4.577-5.005 3.923 0 5.145 5.126 5.423 6.231zM0 7.003C0 11.071 3.06 16.484 12 22c8.94-5.516 12-10.929 12-14.997 0-7.962-9.648-9.028-12-3.737C9.662-1.996 0-1.004 0 7.003z"
              fill="#DD0031"
            />
          </svg>
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <div
        className="cardContent"
      >
        <div
          className="infoSection"
        >
          <a
            href="https://www.8.com/"
            style={
              Object {
                "textDecoration": "none",
              }
            }
            target="_blank"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "margin": "0.1rem 0.5rem 0 0",
                  }
                }
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1.333c3.676 0 6.667 2.991 6.667 6.667 0 3.676-2.991 6.667-6.667 6.667-3.676 0-6.667-2.991-6.667-6.667 0-3.676 2.991-6.667 6.667-6.667zM8 0C3.582 0 0 3.582 0 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zm.667 8V4H7.333v5.333H12V8H8.667z"
                    fill="#606060"
                  />
                </svg>
              </div>
              <p
                className="textStyle6"
              >
                25 years ago by author-8
              </p>
            </div>
            <p
              className="textStyle3"
            >
              title-8
            </p>
          </a>
        </div>
        <div
          className="likeSection"
          data-testid="like-icon"
          onClick={[Function]}
        >
          <svg
            height="22"
            viewBox="0 0 24 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8.229C12.234 7.109 13.547 2 17.382 2 19.602 2 22 3.551 22 7.003c0 3.907-3.627 8.47-10 12.629C5.627 15.473 2 10.91 2 7.003c0-3.484 2.369-5.005 4.577-5.005 3.923 0 5.145 5.126 5.423 6.231zM0 7.003C0 11.071 3.06 16.484 12 22c8.94-5.516 12-10.929 12-14.997 0-7.962-9.648-9.028-12-3.737C9.662-1.996 0-1.004 0 7.003z"
              fill="#DD0031"
            />
          </svg>
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <div
        className="cardContent"
      >
        <div
          className="infoSection"
        >
          <a
            href="https://www.9.com/"
            style={
              Object {
                "textDecoration": "none",
              }
            }
            target="_blank"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "margin": "0.1rem 0.5rem 0 0",
                  }
                }
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1.333c3.676 0 6.667 2.991 6.667 6.667 0 3.676-2.991 6.667-6.667 6.667-3.676 0-6.667-2.991-6.667-6.667 0-3.676 2.991-6.667 6.667-6.667zM8 0C3.582 0 0 3.582 0 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zm.667 8V4H7.333v5.333H12V8H8.667z"
                    fill="#606060"
                  />
                </svg>
              </div>
              <p
                className="textStyle6"
              >
                24 years ago by author-9
              </p>
            </div>
            <p
              className="textStyle3"
            >
              title-9
            </p>
          </a>
        </div>
        <div
          className="likeSection"
          data-testid="like-icon"
          onClick={[Function]}
        >
          <svg
            height="22"
            viewBox="0 0 24 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8.229C12.234 7.109 13.547 2 17.382 2 19.602 2 22 3.551 22 7.003c0 3.907-3.627 8.47-10 12.629C5.627 15.473 2 10.91 2 7.003c0-3.484 2.369-5.005 4.577-5.005 3.923 0 5.145 5.126 5.423 6.231zM0 7.003C0 11.071 3.06 16.484 12 22c8.94-5.516 12-10.929 12-14.997 0-7.962-9.648-9.028-12-3.737C9.662-1.996 0-1.004 0 7.003z"
              fill="#DD0031"
            />
          </svg>
        </div>
      </div>
    </article>
  </section>
</div>
`;
